### ISSUE TITLE
Enhance local development with Edge Examples and Edge CLI

### ISSUE THREAD

**Issue by carlosdtrejo** [enthought/edge-examples#118]
From support ticket [#166](https://edge-support.enthought.com/a/tickets/166?current_tab=details):

> Running the streamlit app created with the Edge CLI. It looks like some dependencies (e.g.., streamlit) is only installed in the Docker image.
> I can see a case where the user wants to run their application locally without going through `edge app build` when they are getting started. It is also likely quicker since you don't have to rebuild the docker image each time you make a modification. 
> As much as possible, it would be nice to include all application dependencies locally for development (there are likely going to be platform specific things that are only available on Linux and in the built image). 
> Running the image is great when you want to check how things will actually work when deployed, but it shouldn't be the only mechanism for development. 

@EdwinWiseOne enhanced support for local development would be beneficial for users. Would this require changes to the Edge CLI?

**Comment by EdwinWiseOne** [comment]
There are two sections in the config file -- "env_deps" are used to build the environment, and "app_deps" for the docker image.  We could automate merging the two via the CLI when making the environment, or manually add the "app_deps" to the environment list in the examples.

The automated merge _might_ be better, so the two dependency sections are still separated by their semantic purpose, in which case it's a CLI change.

**Comment by carlosdtrejo** [comment]
> The automated merge might be better, so the two dependency sections are still separated by their semantic purpose, in which case it's a CLI change.

I agree. Automating the merge of `env_deps` and `app_deps` during environment setup via the CLI is a practical solution. I'll move this to edge-product board.

**Comment by carlosdtrejo** [comment]
Moved to edge-product https://app.zenhub.com/workspaces/edge-5e9ec7416499d5d446625f83/issues/gh/enthought/edge-product/2979
